lib/Sema/CSDiag.cpp

```diff
@@ -845,9 +845,6 @@ diagnoseTypeMemberOnInstanceLookup(Type baseObjTy,
   if (isa<EnumElementDecl>(member))
     Diag.emplace(diagnose(loc, diag::could_not_use_enum_element_on_instance,
                           memberName));
-  else
```

Comment:
I expect new fix to diagnose all of the cases previously diagnosed by CSDiag for `UR_InstanceMemberOnType` as well as `UR_TypeMemberOnInstance`, so `diagnoseTypeMemberOnInstanceLookup` should be completely eliminated as well as portions of `diagnoseUnviableLookupResults`.

Reply:
Oh, I guess I’ll have to modify my class to accept more arguments, like `member`, just like those functions!

Reply:
Yeah, and some of the stuff you can get from constraint system too, since solution gets applied before diagnosing fixes.